modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java

```diff
@@ -125,20 +129,53 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
           throws WorkflowOperationException {
 
     MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
     logger.info("Start speech-to-text workflow operation for media package {}", mediaPackage);
 
+
     ConfiguredTagsAndFlavors tagsAndFlavors = getTagsAndFlavors(workflowInstance,
             Configuration.none, Configuration.one,
             Configuration.many, Configuration.one);
-    MediaPackageElementFlavor sourceFlavor = tagsAndFlavors.getSingleSrcFlavor();
+    List<MediaPackageElementFlavor> sourceFlavor = tagsAndFlavors.getSrcFlavors();
+    List<String> sourceTagList = tagsAndFlavors.getSrcTags();
+
+
+    MediaPackageElementFlavor matchingFlavor = null;
+    if (!sourceFlavor.isEmpty()) {
+      matchingFlavor = sourceFlavor.get(0);
+    }
```

Comment:
How I think `getSingleSrcFlavor` could be used

```suggestion
     MediaPackageElementFlavor sourceFlavor = tagsAndFlavors.getSingleSrcFlavor();
    List<String> sourceTagList = tagsAndFlavors.getSrcTags();
```